src/Gui/Dialogs/DlgAddPropertyVarSet.cpp

```diff
@@ -515,72 +445,92 @@ void DlgAddPropertyVarSet::critical(const QString& title, const QString& text) {
-    propertyItem->setData(data);
-}
-
-void DlgAddPropertyVarSet::addDocumentation() {
-    /* Add the documentation to an existing property.
-     * Note that this method assumes the property exists.
-     *
-     * Since there is no check on documentation (we accept any string), there
-     * is no signal handler for the documentation field.  This method updates
-     * the property that is being added with the text inserted as
-     * documentation/tooltip.
-     *
-     * This function should be called at a late stage, before doing the accept.
-     */
-
+    std::string name = ui->lineEditName->text().toStdString();
     std::string group = comboBoxGroup.currentText().toStdString();
+    std::string type = ui->comboBoxType->currentText().toStdString();
     std::string doc = ui->lineEditToolTip->text().toStdString();
 
-    // performs a check for nullptr
-    App::Property* prop = getPropertyToAdd();
-    varSet->changeDynamicProperty(prop, group.c_str(), doc.c_str());
+    App::Property* prop = nullptr;
+    try {
+        prop = varSet->addDynamicProperty(type.c_str(), name.c_str(),
+                                          group.c_str(), doc.c_str());
+    }
+    catch (Base::Exception& e) {
+        e.ReportException();
+        critical(QObject::tr("Add property"),
+                 QObject::tr("Failed to add property to '%1': %2").arg(
+                         QString::fromLatin1(varSet->getFullName().c_str()),
+                         QString::fromUtf8(e.what())));
+        return false;
+    }
+
+    if (propertyItem) {
+        objectIdentifier = std::make_unique<App::ObjectIdentifier>(*prop);
+        propertyItem->setPropertyData({prop});
+        propertyItem->bind(*objectIdentifier);
+
+        QVariant data = propertyItem->editorData(editor.get());
+        propertyItem->setData(data);
+    }
+
+    return true;
 }
 
-void DlgAddPropertyVarSet::accept()
+void DlgAddPropertyVarSet::closeTransaction(TransactionOption option)
 {
-    addDocumentation();
-    closeTransaction(COMMIT);
-
-    if (ui->checkBoxAdd->isChecked()) {
-        clearEditors();
-        openTransaction();
-        ui->lineEditName->setFocus();
+    if (transactionID == 0) {
         return;
     }
 
-    std::string group = comboBoxGroup.currentText().toStdString();
-    std::string type = ui->comboBoxType->currentText().toStdString();
-    auto paramGroup = App::GetApplication().GetParameterGroupByPath(
-            "User parameter:BaseApp/Preferences/PropertyView");
-    paramGroup->SetASCII("NewPropertyType", type.c_str());
-    paramGroup->SetASCII("NewPropertyGroup", group.c_str());
-    QDialog::accept();
+    App::GetApplication().closeActiveTransaction(static_cast<bool>(option), transactionID);
+    transactionID = 0;
+}
+
+void DlgAddPropertyVarSet::clearFields()
+{
+    bool beforeBlocked = ui->lineEditName->blockSignals(true);
+    ui->lineEditName->clear();
+    ui->lineEditName->blockSignals(beforeBlocked);
```

Comment:
```suggestion
    {
        QSignalBlocker blocker(ui->lineEditName);
        ui->lineEditName->clear();
    }
```

That would be the idiomatic way to achieve that.

Reply:
Yes, I agree.